src/main/java/it/pagopa/pn/externalchannels/pecbysmtp/jmailutils/JMailStoreWrapper.java

```
@@ -0,0 +1,159 @@
+    }
+
+
+    public List<PecEntry> listEntries() {
+        List<PecEntry> pecEntries = new LinkedList<>();
+
+        if(StringUtils.isNotBlank(singleFolder )) {
+            log.info("List IMAP messages for folderName={}", singleFolder);
+            Folder f = getFolderFromStore( this.singleFolder );
+            pecEntries.addAll( listEntries( f ) );
+        }
+        else {
+            log.info("List IMAP messages for all folders");
+            for(Folder f: allStoreFolders() ) {
+                pecEntries.addAll( listEntries( f ) );
+            }
+        }
+
+        return new ArrayList<>( pecEntries );
+    }
+
+    private Folder getFolderFromStore( String folderName) {
+        try {
+            return store.getDefaultFolder().getFolder( folderName );
+        } catch (MessagingException exc) {
+            throw new PnInternalException( "Reading IMAP folder " + folderName, exc );
+        }
+    }
+
+    private List<Folder> allStoreFolders() {
+        List<Folder> allStoreFolders = new ArrayList<>();
+
+        try {
+            log.info("List 'PersonalNamespaces' folder and sub folders");
+            Folder[] personalFolders = store.getPersonalNamespaces();
+            for( Folder f: personalFolders) {
+                allStoreFolders.addAll( walkFolderTree( f ));
+            }
+            allStoreFolders.addAll(Arrays.asList( personalFolders ) );
+        }
+        catch ( MessagingException exc) {
+            log.error("Listing personal folders", exc);
+        }
+
+
+        try {
+            log.info("List 'SharedNamespaces' folder and sub folders");
+            Folder[] sharedFolders = store.getSharedNamespaces();
+            for( Folder f: sharedFolders) {
+                allStoreFolders.addAll( walkFolderTree( f ));
+            }
+            allStoreFolders.addAll(Arrays.asList( sharedFolders ) );
+        }
+        catch ( MessagingException exc) {
+            log.error("Listing shared folders", exc);
```

E' sufficiente un log per questo tipo di errore?

---

Ho aggiuntop il rethrow, nel dubbio è meglio fail fast.